File: Jan/Week 2/LeetCode/problem_2.cpp
//  Day 9 Problem No 18

// 1929. Concatenation of Array => https://leetcode.com/problems/concatenation-of-array/description/

class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int n=nums.size();
        vector<int> ans(2*n);
        for(int i=0;i<n;i++){
            ans[i] = nums[i];
            ans[i+n] = nums[i];
        }
        return ans;
    }
};

File: Jan/Week 3/Extra/problem_2.py
# Day 16 Problem No 35

# 1071. Greatest Common Divisor of Strings => https://leetcode.com/problems/greatest-common-divisor-of-strings/description/?envType=study-plan-v2&envId=leetcode-75

import math
class Solution(object):
    def gcdOfStrings(self, str1, str2):
        
        if str1+str2 != str2+str1:
            return ""

        gcd_len = math.gcd(len(str1),len(str2))

        return str1[:gcd_len]
        